src/core/lib/math/distributiongenerator.cpp

```diff
@@ -35,12 +35,142 @@
+            // versions of GCC in MinGW)
+            std::array<uint32_t, PRNG::MAX_SEED_GENS> initKey{};
+            // high-resolution clock typically has a nanosecond tick period
+            // Arguably this may give up to 32 bits of entropy as the clock gets
+            // recycled every 4.3 seconds
+            initKey[0] = std::chrono::high_resolution_clock::now().time_since_epoch().count();
+            // A thread id is often close to being random (on most systems)
+            initKey[1] = std::hash<std::thread::id>{}(std::this_thread::get_id());
+            // On a 64-bit machine, the thread id is 64 bits long
+            // skip on 32-bit arm architectures
+    #if !defined(__arm__) && !defined(__EMSCRIPTEN__)
+            if (sizeof(size_t) == 8)
+                initKey[2] = (std::hash<std::thread::id>{}(std::this_thread::get_id()) >> 32);
+    #endif
+            // heap variable; we are going to use the least 32 bits of its memory
+            // location as the counter. This will increase the entropy of the PRNG sample
+            void* mem        = malloc(1);
+            uint32_t counter = reinterpret_cast<long long>(mem);  // NOLINT
+            free(mem);
+
+            std::uniform_int_distribution<uint32_t> distribution(0);
+            // the code below is wrapped in to {} as we want to get rid of gen immediately after the loop
+            {
+                // "PRNG* gen" points at a dynamically allocated (using c++'s new()) memory!!!
+                std::unique_ptr<PRNG> gen(genPRNGEngine(initKey, counter));
+                for (auto& s : seed)
+                    s = distribution(*gen);
+            }
+
+            std::array<uint32_t, PRNG::MAX_SEED_GENS> rdseed{};
+            size_t attempts  = 3;
+            bool rdGenPassed = false;
+            for (size_t i = 0; i < attempts && !rdGenPassed; ++i) {
+                try {
+                    std::random_device genR;
+                    for (auto& rds : rdseed) {
+                        // we use the fact that there is no overflow for unsigned integers
+                        // (from C++ standard) i.e., arithmetic mod 2^32 is performed. For
+                        // the seed to be random, it is sufficient for one of the two
+                        // samples below to be random. In almost all practical cases,
+                        // distribution(genR) is random. We add distribution(gen) just in
+                        // case there is an implementation issue with random_device (as in
+                        // older MinGW systems).
+                        rds = distribution(genR);
+                    }
+                    rdGenPassed = true;
+                }
+                catch (std::exception& e) {
+                }
+            }
+            for (uint32_t i = 0; i < PRNG::MAX_SEED_GENS; ++i)
+                seed[i] += rdseed[i];
+  #endif  // FIXED_SEED
```

Comment:
rdseed should be erased here, as this contains the majority of the seed entropy.